crates/hyperswitch_domain_models/src/payments/payment_intent.rs

```diff
@@ -287,39 +293,44 @@ pub enum PaymentIntentUpdate {
         status: storage_enums::IntentStatus,
         updated_by: String,
     },
-}
+    /// UpdateIntent
+    UpdateIntent {
+        amount: Option<MinorUnit>,
+        currency: Option<storage_enums::Currency>,
+        shipping_cost: Option<MinorUnit>,
+        // TODO: Check how to handle this
+        // tax_details: Option<diesel_models::TaxDetails>,
+        skip_external_tax_calculation: Option<super::TaxCalculationOverride>,
+        skip_surcharge_calculation: Option<super::SurchargeCalculationOverride>,
+        surcharge_amount: Option<MinorUnit>,
+        tax_on_surcharge: Option<MinorUnit>,
+        routing_algorithm_id: Option<id_type::RoutingId>,
+        capture_method: Option<common_enums::CaptureMethod>,
+        authentication_type: Option<common_enums::AuthenticationType>,
+        billing_address: Box<Option<Encryptable<Secret<Address>>>>,
```

Comment:
why are we boxing every field. Boxing has implications, since the data is stored on heap in fragments. We can have a struct which will have all the fields that can be updated and then Box that struct. This way there will be only one access to the heap, which will get / store the data from a single place